GithubApiWithVIPER/GithubApiWithVIPERTests/GithubApiWithVIPERTests.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  GithubApiWithVIPERTests.swift
+//  GithubApiWithVIPERTests
+//
+//  Created by Yusuf Özgül on 19.09.2020.
+//  Copyright © 2020 Yusuf Özgül. All rights reserved.
+//
+
+import XCTest
+@testable import GithubApiWithVIPER
+
+/// test cases for repo search module
+class GithubApiWithVIPERTSearchModuleests: XCTestCase {
+    private var presenter: GithubSearchPresenter!
+    private var interactor: GithubSearchInteractor!
+    private var view: MockView!
+    private var router: MockRouter!
+    
+    override func setUp() {
+        interactor = GithubSearchInteractor()
+        view = MockView()
+        router = MockRouter()
+        presenter = .init(view: view, interactor: interactor, router: router)
+        view.presenter = presenter
+    }
+    
+    func testSetPrepareUI() {
+        view.viewdidLoad()
+        XCTAssertEqual(view.prepareUICounts, 1)
+    }
+    
+    func testErroredSearch() {
```

Comment:
Aslında bu biraz style ile alakalı ama test yazdığın method tek bir şeyi test etmeli adı üstünde unit test. Ve neyi test ettiğini method isminde yazarsan, sonradan bakan kişi sadece method isminden neyi test ettiğini anlayabilir. Biz genelde şöyle ilerliyoruz test method naming convension'da

test_methodName_Condition_Behavior

bu durumda burası 
**test_handleSearchResult_Failure_ShouldInvokeShowError**

ayrıca failure durumunda 
view?.showError(with: error.localizedDescription) çağrılmış yani test etmemiz gereken şey ilgili method çağrıldı mı ve doğru errorMessage ile çağrıldı mı onu assert etmek daha uygun olabilir

Aslında sen de benzer şeyi yapmışsın, description assertion'u var ama bir de direk xctfail yerine direk count kontrolü de eklenebilir ama senin yaptığında cover eder gibi duruyor;
            XCTAssertEqual(view.errors.count, 1)
            XCTAssertEqual(view.errors[0], GithubApiError.urlEncode.localizedDescription)

Reply:
Buradaki sıkıntı şu oluyor bu durumda. Eğer count 0 gelirse fail alıyorum haliyle ancak alttaki satırı da test etmek istediği için bu sefer test case crash ediyor.